### PR TITLE
Run tests with Python 3.5.0

### PR DESCRIPTION
**Pull request by Gallaecio**
Related to #4261

### CODE REVIEW

**Comment by elacuesta** [comment]
The patch looks simple and harmless, but it seems like the `3.5.0` version is not available, at least for Ubuntu 16.04 :disappointed: 

```
Downloading archive: https://storage.googleapis.com/travis-ci-language-archives/python/binaries/ubuntu/16.04/x86_64/python-3.5.0.tar.bz2
$ curl -sSf --retry 5 -o python-3.5.0.tar.bz2 ${archive_url}
curl: (22) The requested URL returned error: 404 Not Found
Unable to download 3.5.0 archive
```

**Comment by kmike** [comment]
AFAIK Ubuntu 16.04 provides 3.5.1, so probably this should be the minimal version we support, not 3.5.0.

File: scrapy/__init__.py [hunk 1]
```diff
@@ -14,8 +14,8 @@
 
 # Check minimum required Python version
 import sys
-if sys.version_info < (3, 5):
-    print("Scrapy %s requires Python 3.5" % __version__)
+if sys.version_info < (3, 5, 1):
```
Comment (kmike) [inline scrapy/__init__.py:17]: Hm, what do you think about allowing it to run in 3.5.0? Most likely, it'd work, so this change may break Scrapy for some users.

**Comment by Gallaecio** [comment]
I’m confused. Xenial should be running 3.5.1 according to [packages.ubuntu.com](https://packages.ubuntu.com/search?suite=xenial&arch=any&searchon=names&keywords=python3), but Travis’ Xenial uses 3.5.2.

File: setup.py [hunk 2]
```diff
@@ -66,7 +66,7 @@ def has_environment_marker_platform_impl_support():
         'Topic :: Software Development :: Libraries :: Application Frameworks',
         'Topic :: Software Development :: Libraries :: Python Modules',
     ],
-    python_requires='>=3.5',
+    python_requires='>=3.5.1',
```
Comment (kmike) [inline setup.py:69]: this would prevent installing Scrapy on 3.5.0 - do we want it?

**Comment by kmike** [comment]
Thanks @Gallaecio!

**Pull request merged**